src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -279,37 +292,74 @@ public User getProxy() {
 
   Loan withProxy(User newProxy) {
     return new Loan(representation, item, user, newProxy, checkinServicePoint,
-      checkoutServicePoint, originalDueDate, loanPolicy, accounts);
+      checkoutServicePoint, originalDueDate, loanPolicy, overdueFinePolicy,
+      lostItemPolicy, accounts);
   }
 
   Loan withCheckinServicePoint(ServicePoint newCheckinServicePoint) {
     return new Loan(representation, item, user, proxy, newCheckinServicePoint,
-      checkoutServicePoint, originalDueDate, loanPolicy, accounts);
+      checkoutServicePoint, originalDueDate, loanPolicy, overdueFinePolicy,
+      lostItemPolicy, accounts);
   }
 
   Loan withCheckoutServicePoint(ServicePoint newCheckoutServicePoint) {
     return new Loan(representation, item, user, proxy, checkinServicePoint,
-      newCheckoutServicePoint, originalDueDate, loanPolicy, accounts);
+      newCheckoutServicePoint, originalDueDate, loanPolicy, overdueFinePolicy,
+      lostItemPolicy, accounts);
   }
 
   public Loan withAccounts(Collection<Account> newAccounts) {
     return new Loan(representation, item, user, proxy, checkinServicePoint,
-      checkoutServicePoint, originalDueDate, loanPolicy, newAccounts);
+      checkoutServicePoint, originalDueDate, loanPolicy, overdueFinePolicy,
+      lostItemPolicy, newAccounts);
+  }
+
+  public Loan withLoanPolicy(LoanPolicy newloanPolicy) {
+    requireNonNull(newloanPolicy, "newloanPolicy cannot be null");
+
+    return new Loan(representation, item, user, proxy, checkinServicePoint,
+      checkoutServicePoint, originalDueDate, newloanPolicy, overdueFinePolicy,
+      lostItemPolicy, accounts);
+  }
+
+  public Loan withOverdueFinePolicy(OverdueFinePolicy newOverdueFinePolicy) {
+    requireNonNull(newOverdueFinePolicy, "newloanPolicy cannot be null");
+
+    return new Loan(representation, item, user, proxy, checkinServicePoint,
+      checkoutServicePoint, originalDueDate, loanPolicy, newOverdueFinePolicy,
+      lostItemPolicy, accounts);
+  }
+
+  public Loan withLostItemPolicy(LostItemPolicy newLostItemPolicy) {
+    requireNonNull(newLostItemPolicy, "newloanPolicy cannot be null");
+
+    return new Loan(representation, item, user, proxy, checkinServicePoint,
+      checkoutServicePoint, originalDueDate, loanPolicy, overdueFinePolicy,
+      newLostItemPolicy, accounts);
   }
 
   public String getLoanPolicyId() {
     return representation.getString("loanPolicyId");
   }
 
+  public String getOverdueFinePolicyId() {
+    return representation.getString("overdueFinePolicyId");
+  }
```

Comment:
Given that the storage representation uses the policy objects when updating, what are these properties used for?

Reply:
This method is used in OverdueFinePolicyRepository class to get overdueFinePolicyId field from loan

Reply:
Ok, another way of doing this would be to create the unknown policy with the ID from the JSON and use that in these methods.

That would mean we could stop referring to the JSON directly in these methods.

Reply:
Could you explain this way in more details?

Reply:
When the loan is created, if a policy hasn't be defined yet, the `unknown` variant of that class is used.

When the loan is first created, the unknown policy classes could be instantiated with an `ID` from the loan JSON.

Then the methods which get that ID could use the appropriate policy field and get the ID from that object.

This would mean that the JSON was no longer needed for those methods, and it would be the same whether the policy had been loaded or not.

Does that make more sense?

Reply:
Fixed